### PR TITLE
Update style-guide.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Minor formatting fixes in intro and minor formatting changes in components list.

### CODE REVIEW

File: source/developer/style-guide.md [hunk 1]
```diff
@@ -127,18 +125,18 @@ function wrongGetStr(stuff) {
 
 ## React-JSX
 
-Part of the build process is running ESLint. ESLint is the final authority on all style issues. PRs will not be accepted unless there are no errors running ESLint. The ESLint configuration file can be found in: [webapp/.eslintrc](https://github.com/mattermost/mattermost-webapp/blob/master/.eslintrc.json)
+Part of the build process is running ESLint. ESLint is the final authority on all style issues. PRs will not be accepted unless there are no errors running ESLint. The ESLint configuration file can be found in: [webapp/.eslintrc](https://github.com/mattermost/mattermost-webapp/blob/master/.eslintrc.json).
```
Comment (justinegeffen) [inline source/developer/style-guide.md]:
```suggestion
Part of the build process is running ESLint. ESLint is the final authority on all style issues. PRs will only be accepted if there are no errors running ESLint. The ESLint configuration file can be found in: [webapp/.eslintrc](https://github.com/mattermost/mattermost-webapp/blob/master/.eslintrc.json).
```

File: source/developer/style-guide.md [hunk 2]
```diff
@@ -127,18 +125,18 @@ function wrongGetStr(stuff) {
...
-You can run eslint using the makefile by using `make check-style`
+You can run ESlint using the makefile by using `make check-style`.
 
 The following is a subset of what ESLint checks for. ESLint is always the authority. 
 
 ### General
 
 - Include only one React component per file.
-- Use class \<name\> extends React.Component over React.createClass unless you need mixins
+- Use class \<name\> extends React.Component over React.createClass unless you need mixins.
```
Comment (justinegeffen) [inline source/developer/style-guide.md]:
```suggestion
- Use class \<name\> extends `React.Component` over `React.createClass` unless you need mixins.
```

I'm unsure whether I've formatted this correctly or whether it should be "Use `class \<name\> extends React.Component`.

File: source/developer/style-guide.md [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
 1. [Go](#go)
 2. [Javascript](#javascript)
 3. [React-JSX](#react-jsx)
 
 ## Go
 
-All go code must follow the golang official [Style Guide](https://golang.org/doc/effective_go.html)
-
-In addition all code must be run though the official go formatter tool [gofmt](https://golang.org/cmd/gofmt/)
+All go code must follow the golang official [Style Guide](https://golang.org/doc/effective_go.html). In addition all code must be run though the official go formatter tool [gofmt](https://golang.org/cmd/gofmt/).
```
Comment (amyblais) [inline source/developer/style-guide.md]:
```suggestion
All go code must follow the golang official [Style Guide](https://golang.org/doc/effective_go.html). In addition, all code must be run though the official go formatter tool [gofmt](https://golang.org/cmd/gofmt/).
```

File: source/developer/style-guide.md (hunk 1 above)
Comment (amyblais) [reply source/developer/style-guide.md]: Agree,

File: source/developer/style-guide.md (hunk 2 above)
Comment (jasonblais) [reply source/developer/style-guide.md]: I'm not entirely sure, but your's looks correct. I wonder if the word `extends` is meant to be there

Comment (justinegeffen) [reply source/developer/style-guide.md]: I was also wondering that but am not familiar enough with the language to make the call. I've looked up the areas of expertise and perhaps @saturninoabril can help? :)

**Comment by justinegeffen** [comment]
@saturninoabril, would you be able to assist with confirming some of the language used? :)

File: source/developer/style-guide.md [hunk 4]
```diff
@@ -82,7 +80,7 @@ var my_variable = 4;
 
 - Braces must be used on all blocks.
 - Braces must start on the same line as the statement starting the block.
-- Else and else if must be on the same line as the if block closing brace.
+- `Else` and `else if` must be on the same line as the `if` block closing brace.
```
Comment (saturninoabril) [inline source/developer/style-guide.md]:
```suggestion
- `else` and `else if` must be on the same line as the `if` block closing brace.
```
It's Javascript specific keyword.

File: source/developer/style-guide.md [hunk 5]
```diff
@@ -127,18 +125,18 @@ function wrongGetStr(stuff) {
...
 ## React-JSX
 
-Part of the build process is running ESLint. ESLint is the final authority on all style issues. PRs will not be accepted unless there are no errors running ESLint. The ESLint configuration file can be found in: [webapp/.eslintrc](https://github.com/mattermost/mattermost-webapp/blob/master/.eslintrc.json)
+Part of the build process is running ESLint. ESLint is the final authority on all style issues. PRs will be accepted provided there are no errors running ESLint. The ESLint configuration file can be found in: [webapp/.eslintrc](https://github.com/mattermost/mattermost-webapp/blob/master/.eslintrc.json).
 
-Instructions on how to use ESLint with your favourite editor can be found here: [http://eslint.org/docs/user-guide/integrations](http://eslint.org/docs/user-guide/integrations)
+Instructions on how to use ESLint with your favourite editor can be found here: [http://eslint.org/docs/user-guide/integrations](http://eslint.org/docs/user-guide/integrations).
 
-You can run eslint using the makefile by using `make check-style`
+You can run ESlint using the makefile by using `make check-style`.
```
Comment (saturninoabril) [inline source/developer/style-guide.md]:
```suggestion
You can run ESLint using the makefile by using `make check-style`.
```

File: source/developer/style-guide.md [hunk 6]
```diff
@@ -1,16 +1,14 @@
...
 1. [Go](#go)
 2. [Javascript](#javascript)
 3. [React-JSX](#react-jsx)
 
 ## Go
 
-All go code must follow the golang official [Style Guide](https://golang.org/doc/effective_go.html)
-
-In addition all code must be run though the official go formatter tool [gofmt](https://golang.org/cmd/gofmt/)
+All go code must follow the golang official [Style Guide](https://golang.org/doc/effective_go.html). In addition, all code must be run though the official go formatter tool [gofmt](https://golang.org/cmd/gofmt/).
```
Comment (saturninoabril) [inline source/developer/style-guide.md]:
```suggestion
All Go code must follow the Golang official [Style Guide](https://golang.org/doc/effective_go.html). In addition, all code must be run though the official Go formatter tool [gofmt](https://golang.org/cmd/gofmt/).
```

File: source/developer/style-guide.md (hunk 2 above)
Comment (saturninoabril) [reply source/developer/style-guide.md]: Yup, it should be there, actually starting with class.
```suggestion
- Use `class <Name> extends React.Component` over `React.createClass` unless you need mixins.
```

**Comment by justinegeffen** [comment]
Thanks very much @saturninoabril! Much appreciated. :)

**Pull request merged**